src/deprecated.jl

```diff
@@ -599,8 +599,14 @@ end
         return missing
     else
         @inbounds pool = source.pools[col]
-        return get🐱(pool, str::Tuple{Ptr{UInt8}, Int})
-    end
+        p, len = str::Tuple{Ptr{UInt8}, Int}
+        level = unsafe_string(p, len)
```

Comment:
@quinnj Would it be more efficient to use a `WeakRefString` or something like this here? We just need to be able to hash the object and call `==` on it.

Reply:
See [here](https://github.com/JuliaData/Parsers.jl/blob/master/src/strings.jl#L19) for why this works. Can we use the old code here that checked `ht_keyindex2`; it's much faster than doing two `get` calls.

Reply:
Mmm, except that it didn't... :-/ The test I added failed, that's why I rewrote this.

Anyway, that's severe type piracy! I can try to make this work without type piracy using a custom wrapper struct if you want.

Reply:
Personally, I think people are way too heretical about type piracy; this is far from severe. It's an obscure type that's rarely used; if someone is dealing w/ a `Tuple{Ptr{UInt8}, Int}`, they'd probably _want_ this behavior anyway.

I'd have to look into why it didn't work in this case.

Reply:
OK, got it. Calling `push_level!` is not enough, that's an internal function which is supposed to be combined with an insertion into `invindex`. I don't like calling these non-public functions, so I've changed it to `get` and `get!`. We can always add an API to CategoricalArrays to avoid the double lookup, but for now it should be fast enough.

So I've reverted to passing the tuple directly for the lookup. But that type piracy should really be fixed. Also (and maybe more problematically), the `isequal` definition is vulnerable to hash collisions, which isn't acceptable when dealing with user data.

Reply:
Why not use `WeakRefString` instead of a tuple?

Reply:
We can certainly use WeakRefString; that's the right call here.